modules/orchestrator/services/runtime/runtime.go

```diff
@@ -1688,7 +1689,12 @@ func (r *Runtime) FullGC() {
 		}
 	}()
 
-	keep := 5
+	rollbackCfg, err := r.getRollbackConfig()
```

Comment:
it's high dangerous to query all projects in memory.

Reply:
I checked the actual data and there is no more than 300. This function will not calculate other data in the project, such as issue. So the risk is not high for now. 

The temporary cache solution feels more risky, we can do detailed cache solution later in the future.

Reply:
please mark a comment TODO or FIXME, at least

Reply:
done